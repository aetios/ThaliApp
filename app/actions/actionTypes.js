export const NAVIGATE = 'NAVIGATE';
export const BACK = 'BACK';
export const OPENDRAWER = 'OPENDRAWER';
export const CALENDARRETREIVED = 'CALENDARRETREIVED';
export const CALENDARERROR = 'CALENDARERROR';
export const RESETLOGINSTATE = 'RESETLOGINSTATE';
export const WELCOME = 'WELCOME';
