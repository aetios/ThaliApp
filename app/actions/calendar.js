/**
 * Created by aucke on 3/8/17.
 */
