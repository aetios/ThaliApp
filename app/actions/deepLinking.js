export const DEEPLINK = 'DEEPLINKING_DEEPLINK';

export function deepLink(url, stayInApp = true) {
  return {
    type: DEEPLINK,
    payload: { url, stayInApp },
  };
}
