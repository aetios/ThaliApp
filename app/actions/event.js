export const EVENT = 'EVENT_EVENT';
export const FETCHING = 'EVENT_FETCHING';
export const SUCCESS = 'EVENT_SUCCESS';
export const FAILURE = 'EVENT_FAILURE';

export function event(pk, token) {
  return {
    type: EVENT,
    payload: { pk, token },
  };
}

export function success(eventData, eventRegistrations) {
  return {
    type: SUCCESS,
    payload: { eventData, eventRegistrations },
  };
}

export function fetching() {
  return {
    type: FETCHING,
  };
}

export function failure() {
  return {
    type: FAILURE,
  };
}
