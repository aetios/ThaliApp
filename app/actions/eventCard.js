import * as types from "./actionTypes";
export function openEvent(eventId) {
  console.log("opening event");
  return {
  };
}
