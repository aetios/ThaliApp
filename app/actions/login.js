import { AsyncStorage } from 'react-native';
import * as types from './actionTypes';

const USERNAMEKEY = '@MyStore:username';
const TOKENKEY = '@MyStore:token';

export function loginSuccess(username, token) {
  return {
    type: types.LOGINSUCCESS,
    loginState: 'loggedIn',
    username,
    token,
  };
}

export function loginProgress() {
  return {
    type: types.LOGINPROGRESS,
    loginState: 'progress',
  };
}

export function loginFailure() {
  return {
    type: types.LOGINFAILURE,
    loginState: 'failure',
  };
}

export function login(username, password) {
  return (dispatch) => {
    dispatch(loginProgress());
    const data = {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        username,
        password,
      }),
    };
    return fetch('http://localhost:8000/api/token-auth/', data)
      .then(
        response => response.json())
      .then(
        (responseJson) => {
          if (responseJson.token) {
            return AsyncStorage.multiSet([[USERNAMEKEY, username], [TOKENKEY, responseJson.token]])
              .then(
                () => dispatch(loginSuccess(username, responseJson.token)),
              );
          }
          return dispatch(loginFailure());
        })
      .catch(() => dispatch(loginFailure()));
  };
}

export function logout() {
  AsyncStorage.multiRemove([USERNAMEKEY, TOKENKEY]);
  return {
    type: types.LOGOUT,
  };
}
