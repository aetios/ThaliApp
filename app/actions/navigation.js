import * as types from './actionTypes';

export function navigate(scene, newSection = false) {
  return {
    type: types.NAVIGATE,
    scene,
    newSection,
  };
}

export function back() {
  return {
    type: types.BACK,
  };
}

export function updateDrawer(drawerState) {
  return {
    type: types.OPENDRAWER,
    drawerOpen: drawerState,
  };
}
