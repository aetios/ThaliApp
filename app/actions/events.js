import * as types from './actionTypes';
import { navigate } from './navigation';
import { apiUrl } from '../url';

export function start() {
  return {
    type: types.LOADEVENTSTART,
  };
}

export function success(data, registrations) {
  return {
    type: types.LOADEVENTSUCCESS,
    data,
    registrations,
  };
}

export function fail() {
  return {
    type: types.LOADEVENTFAILURE,
  };
}

function loadRegistrations(id, token) {
  const data = {
    method: 'GET',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      Authorization: `Token ${token}`,
    },
  };
  return fetch(`${apiUrl}/events/${id}/registrations/?status=registered`, data)
    .then(
      response => response.json(),
    )
    .catch(
      () => [],
    );
}


export function loadEvent(id, token) {
  return (dispatch) => {
    dispatch(start());
    dispatch(navigate('event'));

    const data = {
      method: 'GET',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        Authorization: `Token ${token}`,
      },
    };
    return fetch(`${apiUrl}/events/${id}/`, data)
      .then(
        response => response.json(),
      )
      .then(
        (response) => {
          if (response.status > -1) {
            loadRegistrations(id, token)
              .then(registrations => dispatch(success(response, registrations)));
          } else {
            dispatch(success(response, []));
          }
        },
      )
      .catch(
        () => dispatch(fail())
      );
  };
}
