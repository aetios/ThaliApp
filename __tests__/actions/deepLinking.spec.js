import * as actions from '../../app/actions/deepLinking';

describe('deeplinking actions', () => {
  it('should expose the deeplinking actions', () => {
    expect(actions.DEEPLINK).toEqual('DEEPLINKING_DEEPLINK');
  });

  it('should create an action to notify of a new deeplink', () => {
    expect(actions.deepLink('http://example.org')).toMatchSnapshot();
  });
});