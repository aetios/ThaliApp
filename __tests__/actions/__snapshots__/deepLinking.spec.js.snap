// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deeplinking actions should create an action to notify of a new deeplink 1`] = `
Object {
  "payload": Object {
    "url": "http://example.org",
  },
  "type": "DEEPLINKING_DEEPLINK",
}
`;
